### ISSUE TITLE
Shutdown and reboot buttons not working

### ISSUE THREAD

**Issue by turion** [sddm/sddm#458]
I'm using sddm for KDE 5 on Gentoo. The shutdown and reboot buttons on the login screen don't work for me. I'm also affected by #405, maybe they're related. Can I provide any useful logs or config files?

**Comment by saikatbsk** [comment]
Similar issue here. The shutdown and reboot buttons do not work in sddm. I am using plasma 5.7 on gentoo.

**Comment by WiseLord** [comment]
I'm using gentoo with openrc. xdm is added to default runlevel and /etc/conf.d/xdm configured to use sddm.
When system boots, reboot/shutdown buttons don't work. But if I restart xdm manually (/etc/init.d/sddm restart), buttons works.
In both cases (when buttons work and when no) I see in /var/log/sddm.log, that buttons send messages:

[18:43:12.033] (II) DAEMON: Message received from greeter: PowerOff
[18:43:30.478] (II) DAEMON: Message received from greeter: Reboot

**Comment by WiseLord** [comment]
Just realized that bug dissappears in my case if 'consolekit' is added to 'default' runlevel in gentoo.

**Comment by vokiel-too** [comment]
They don't even show up for me and btw consolekit has been removed some time ago.

**Comment by larryljordan** [comment]
I am using Artix with Plasma 5 and OpenRC the shutdown and reboot buttons worked on fresh install but at some point they stopped working.

**Comment by grahamperrin** [comment]
Duplicate of, or overlap with, #434

**Comment by Fitzcarraldo1** [comment]
My Gentoo Linux uses OpenRC, not systemd. The Sleep, Restart and Shut Down buttons on the SDDM greeter screen were greyed out and not working until I added elogind to the boot run level:

root # rc-update add elogind boot

Once elogind is added to the boot run level so that it runs before the display manager, the Sleep, Restart and Shut Down buttons on the greeter screen are no longer greyed out and they function as expected.

**Comment by Vogtinator** [comment]
Is elogind configured to be dbus activatable?

**Comment by Fitzcarraldo1** [comment]
> Is elogind configured to be dbus activatable?

Yes. When I merged sys-auth/elogind-252.9 the ebuild added the following elog message to the build log file:

"elogind is currently not started from any runlevel.
You may add it to the boot runlevel by:

# rc-update add elogind boot

Alternatively, you can leave elogind out of any
runlevel. It will then be started automatically
when the first service calls it via dbus, or
the first user logs into the system."

**Comment by Vogtinator** [comment]
In that case it's probably `LogindPathInternal::LogindPathInternal` that needs adjusting. It currently uses `isServiceRegistered` but would need to do a blocking `startService` instead. PRs welcome.
